packages/api-gateway-service/service.ts

```diff
@@ -75,31 +116,38 @@ const server = http.createServer( app );
   apollo.installSubscriptionHandlers( server );
 } )();
 
-/* Auth Token Verification Check */
-app.post( '/graphql', ( req, res, next ) => {
-  if ( !req.headers.authorization ) {
-    return res.status( 401 ).json( new AuthenticationError( 'Auth Token Missing' ) );
+
+const getAuthenticatedUserDetails =  (req: express.Request) => {
+  const cred = basicAuth(req) ? basicAuth(req) : null;
+  const user = cred ? { name: cred.name, password: cred.pass } : { name: null, pass: null };
+  const validUsers = { name: process.env.API_USER, password:process.env.API_PASS };
+  if( JSON.stringify(user) === JSON.stringify(validUsers)) {
+    return user;
   } else {
-    publicKey().then( ( key: string ) => {
-      try {
-        const tokenArray: any = req.headers.authorization?.split( ` ` );
-        const accessToken = tokenArray[ tokenArray.length - 1 ] || req.cookies[ 'access-token' ];
-        verify( accessToken, key, { algorithms: [ 'RS256' ] }, ( err: any, payload: any ) => {
-          if ( err && err.name === 'TokenExpiredError' ) {
-            return res.status( 403 ).json( err );
-          } else if ( err && err.name === 'JsonWebTokenError' ) {
-            return res.status( 403 ).json( err );
-          } else if ( !err ) {
-            req.headers = { ...req.headers, tokenPaylod: JSON.stringify( payload ) };
-            next();
-          }
-        } );
-      } catch ( err ) {
-        return res.status( 403 ).json( err );
-      }
-    } );
+    return null;
   }
-} );
+};
+
+app.get('/graphql', (req, res, next) => {
```

Comment:
Why is the auth middleware just applied to only GET requests??

Should use `.use` to make sure it gets applied to all kinds of request methods.

```suggestion
app.use('/graphql', (req, res, next) => {
```

Reply:
I made it as get for accessing the playground. if you are using the postman/from portal both way of auth will work